filebeat/input/filestream/input.go

```diff
@@ -591,42 +671,65 @@ func (inp *filestream) readFromSource(
 			} else {
 				log.Errorf("Read line error: %v", err)
 				metrics.ProcessingErrors.Inc()
+				if isGZIP {
+					metrics.ProcessingGZIPErrors.Inc()
+				}
 			}
 
 			return nil
 		}
 
+		// sate offset increase
 		s.Offset += int64(message.Bytes) + int64(message.Offset)
 
 		flags, err := message.Fields.GetValue("log.flags")
 		if err == nil {
 			if flags, ok := flags.([]string); ok {
 				if slices.Contains(flags, "truncated") { //nolint:typecheck,nolintlint // linter fails to infer generics
 					metrics.MessagesTruncated.Add(1)
+					if isGZIP {
+						metrics.MessagesGZIPTruncated.Add(1)
+					}
```

Comment:
We consider GZIP files as static. But as we have the metric foe plain files, I left it here fro completeness